src/js/components/tasks/submit.tsx

```diff
@@ -15,17 +15,19 @@ import {
 import { OBJECT_TYPES } from '@/utils/constants';
 
 interface Props {
-  taskId: string;
-  taskName: string;
-  taskDiffUrl: string | null;
+  instanceId: string;
+  instanceName: string;
+  instanceDiffUrl: string | null;
+  instanceType: 'task' | 'project';
   isOpen: boolean;
   toggleModal: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
 const SubmitModal = ({
-  taskId,
-  taskName,
-  taskDiffUrl,
+  instanceId,
+  instanceName,
+  instanceDiffUrl,
+  instanceType,
```

Comment:
Looks like there are quite a few instances of `"task"` in this file that should change?

Reply:
I think it should also be moved out into `utils` or similar, yeah?

Reply:
Yes, agreed. Though it's really just shared between these two modules (projects and tasks), and it's large enough to deserve its own file... Maybe `utils.tsx` should also get split into multiple files in a new `components/utils/` dir.

Reply:
On it, unless you already are!